cli/cmd/bb/bb.go

```diff
@@ -2,29 +2,56 @@ package main
 
 import (
 	"flag"
+	"os"
 
 	"github.com/buildbuddy-io/buildbuddy/cli/arg"
 	"github.com/buildbuddy-io/buildbuddy/cli/bazelisk"
+	"github.com/buildbuddy-io/buildbuddy/cli/log"
 	"github.com/buildbuddy-io/buildbuddy/cli/login"
 	"github.com/buildbuddy-io/buildbuddy/cli/parser"
+	"github.com/buildbuddy-io/buildbuddy/cli/plugin"
 	"github.com/buildbuddy-io/buildbuddy/cli/remotebazel"
 	"github.com/buildbuddy-io/buildbuddy/cli/sidecar"
+	"github.com/buildbuddy-io/buildbuddy/cli/terminal"
 	"github.com/buildbuddy-io/buildbuddy/cli/tooltag"
 	"github.com/buildbuddy-io/buildbuddy/cli/version"
 )
 
 func main() {
 	flag.Parse()
+	exitCode, err := run()
```

Comment:
instead of pulling this out into a `run()` function, what do you think leaving the logic in main() and bringing back this die method?

https://github.com/buildbuddy-io/buildbuddy/commit/a83ffc7e271088a027e5c186b143144523cc27dd#diff-0ddf6d0f5a686b3782a8c089af99f308f250aeea9741808012cd92e9c4e98f5dL31

Reply:
I think the control flow is less error-prone if we have only one place in the program that can call `exit()`.

Usually we can look at a `defer` statement and say "this should get executed when the function returns," but if some function calls potentially call `exit` then we can no longer rely on that. So it's hard to tell whether a particular cleanup func will actually get called or not. (Even more basic than `defer`: we can usually look at a function call and say "the code after this call should execute once this function returns")

I ran into this when writing that `defer` in the block below -- it wasn't getting called and I was scratching my head.